freqtrade/strategy/interface.py

```diff
@@ -362,9 +362,9 @@ def stop_loss_reached(self, current_rate: float, trade: Trade,
             # Don't update stoploss if trailing_only_offset_is_reached is true.
             if not (self.trailing_only_offset_is_reached and high_profit < sl_offset):
                 # Specific handling for trailing_stop_positive
-                if 'trailing_stop_positive' in self.config and high_profit > sl_offset:
+                if 'trailing_stop_positive' in self.__dict__ and high_profit > sl_offset:
```

Comment:
Do we need this check? 
`if 'trailing_stop_positive' in self.__dict__`

It's a class attribute and this check will always return true...

Reply:
nope - it's a class attribute which has no value assigned => does not exist at instance level unless it's overwritten.

I've tested this ... just removing this will break 3 or 4 tests :D 

The problem with `trailing_stop_positive` is that it's existence is significant - so if this is missing from both config and strategy, then it's disabled, otherwise that value will be used (also 0.0 is valid).

that's a design i don't particularly like, but see out of scope for this PR to fix ...

Reply:
I just hate this kind of 'code smells' in python, when simple things require using ugly looking code like access to `__dict__`.

Please review the changes I made to fix it.

Reply:
Yes, this works as well.

Honestly, i didn't spend much time to find a better solution since as mentioned above, i think this setting should be explicit (`trailing_stop_positive_enabled` or similar), and i intend to change it to that very soon.

I've swapped the comments in these 2 lines - since the one you removed was correct (and makes sense) - but the one a line below does not (since this changed now).